File: src/main/java/com/harshit/pharmacy/user/service/UserService.java
package com.harshit.pharmacy.user.service;

import com.harshit.pharmacy.user.record.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UserService {

    AdminUserResponse createUser(CreateUserRequest request);

    Page<AdminUserResponse> getAllUsers(Pageable pageable);

    Page<AdminUserResponse> searchUsers(String email, Pageable pageable);

    AdminUserResponse updateUserStatus(Integer userId, UserStatusRequest request);

    UserProfileResponse getMyProfile();

    UserProfileResponse updateProfile(UserProfileRequest request);

    void changeEmail(ChangeEmailRequest request);

    void changePassword(ChangePasswordRequest request);
}

File: pharmacy-frontend/src/services/userService.js
import api from "../api/api";

export async function getMyProfile() {
    const response = await api.get("/users/me");
    return response.data;
}

export async function updateProfile(data) {
    const response = await api.put("/users/profile", data);
    return response.data;
}

export async function changeEmail(data) {
    const response = await api.patch("/users/me/change-email", data);
    return response.data;
}

export async function changePassword(data) {
    const response = await api.patch("/users/me/change-password", data);
    return response.data;
}